mlir/docs/Canonicalization.md

```diff
@@ -51,6 +56,60 @@ Some important things to think about w.r.t. canonicalization patterns:
 *   It is always good to eliminate operations entirely when possible, e.g. by
     folding known identities (like "x + 0 = x").
 
+*   Canonicalize isn't a great place to put pattens with expensive compile time
```

Comment:
nit: `Canonicalization`. Also maybe `running time` is a better term because we are talking about the running time of the compiler. (Not the compile time of the C++ pattern implementation.)